apps/aecore/src/aec_peer_connection.erl

```diff
@@ -149,14 +148,32 @@ handle_cast(give_tcp_socket_ownership, S0 = #{ role := responder, tcp_sock := Tc
             %% Report this to aec_peers!? And possibly fail?
             %% Or, we can't do this yet we don't know the port?!
             TRef = erlang:start_timer(?FIRST_PING_TIMEOUT, self(), first_ping_timeout),
-            {noreply, S#{ status => {connected, ESock}, r_pubkey => RemotePub,
-                          first_ping_tref => TRef }};
+            S1 = S#{ status => {connected, ESock}
+                   , r_pubkey => RemotePub
+                   , first_ping_tref => TRef },
+            gen_server:enter_loop(?MODULE, [], S1);
         {error, Reason} ->
             %% What to do here? Close the socket and stop?
             lager:info("Connection accept failed - ~p was from ~p", [Reason, maps:get(host, S)]),
-            gen_tcp:close(TcpSock),
-            {stop, normal, S}
-    end;
+            gen_tcp:close(TcpSock)
+    end.
+
+%% Called when connecting to a peer
+init([Opts]) ->
+    Version = <<?P2P_PROTOCOL_VSN:64>>,
+    Genesis = aec_chain:genesis_hash(),
+    Opts1 = Opts#{ role => initiator
+                   , version => Version
+                   , genesis => Genesis},
+    {ok, Opts1, 0}.
+
+handle_call(Request, From, State) ->
+    handle_request(State, Request, From).
+
+handle_cast(retry, S = #{ host := Host, port := Port, status := error }) ->
+    Self = self(),
+    Pid = spawn(fun() -> try_connect(Self, Host, Port, 1000) end),
```

Comment:
It looks like the gen_tcp:connect embedded in this try_connect call has the possibility of passing send_timeout option. Shall that be configurable?

Reply:
I don't know, as I said before, I though outgoing wasn't part of the ticket.